### PR TITLE
adds ability to mark students as inactive in a given group, and hides…

### PR DESCRIPTION
**Pull request by adambray**
… inactive students by default on attendance, assignments, and todo areas

Note: there are no feature specs for this yet... but feel free to add some! Or I can do that on Wed before we merge this in.

### CODE REVIEW

File: app/controllers/assignments_controller.rb [hunk 1]
```diff
@@ -11,17 +13,18 @@ def index
...
-    @current_user_is_admin = @assignment.group.has_admin?(current_user)
-    redirect_to(current_user, flash:{alert: "You're not authorized."}) if !@current_user_is_admin
-    @submissions = @assignment.submissions
-    if params[:group]
-      @group = Group.at_path(params[:group])
-      @submissions = @submissions.select{|s| s.user.is_member_of(@group)}
-    else
-      @group = @assignment.group
+    @group = Group.at_path(params[:group]) || @assignment.group
+    @submissions = @assignment.submissions.select{|s| s.user.is_member_of(@group) }
```
Comment (RobertAKARobin) [inline app/controllers/assignments_controller.rb]: You could save a whole 0.14 seconds by using `.includes`:
```
@assignment.submissions.includes(:user).select{|s| s.user.is_member_of(@group)}
```

File: app/controllers/assignments_controller.rb [hunk 2]
```diff
@@ -11,17 +13,18 @@ def index
...
+    @submissions.sort_by!{|s| s.user.last_name}
+
+    @show_inactive = params[:show_inactive] == "true"
+    @show_na = params[:show_na] == "true"
+
+    unless @show_inactive
+      @submissions = @submissions.select{|s| s.user.memberships.find_by(group: @group).active? }
```
Comment (RobertAKARobin) [inline app/controllers/assignments_controller.rb]:
```rb
@submissions.includes(user: :memberships).select{|s| s.user.memberships.find_by(group: @group).active? }
```

File: app/views/groups/show.html.erb [hunk 3]
```diff
@@ -36,8 +39,22 @@
...
               <%= avg_obz %>
             </td>
-            <td><%= link_to("Remove", group_membership_path(@group, user), method: :delete, data: {confirm: "Are you sure?"}) if @member_ids.include?(user.id) %></td>
+
+            <% if @member_ids.include?(user.id) %>
+              <td>
+                <%= link_to("Remove", group_membership_path(@group, user), method: :delete, data: {confirm: "Are you sure?"}) %>
+              </td>
+              <td>
+                <%  toggle_active_link_text = user.memberships.find_by(group: @group).active? ? "Deactivate" : "Reactivate" %>
```
Comment (RobertAKARobin) [inline app/views/groups/show.html.erb]: Maybe "Activate" instead of "Reactivate"? Visually looks super-similar.

**Comment by jshawl** [comment]
:+1:

**Comment by RobertAKARobin** [comment]
:+1: pending comments.

**Comment by adambray** [comment]
@RobertAKARobin check out my latest commits! They manage to get the relevant stuff down to 1 FREAKIN Query using `includes`, `references`, and `where` clauses. Additionally, I use a query builder approach to still limit to one query instead of using `select` in ruby-land.

The new version serves the request in ~200ms, vs ~700ms with the previous version!

**Comment by adambray** [comment]
Also, note that I modified the Group model to behave more sanely if given a nil value for `path_at`... let me know if you disagree...

**Comment by adambray** [comment]
PPS - plan to write tests after I grade P2.

**Comment by RobertAKARobin** [comment]
Noooice. Color me impressed!

**Comment by RobertAKARobin** [comment]
Two thumbs! Mergin'.

**Pull request merged**